src/game/server/client.cpp

```diff
@@ -588,6 +589,124 @@ void SV_CreateClientCommands()
+
+	g_ClientCommands.Create("cheat_infiniteair", [](CBasePlayer* player, const CommandArgs& args)
+		{ player->ToggleCheat(Cheat::InfiniteAir); },
+		{.Flags = ClientCommandFlag::Cheat});
+
+	g_ClientCommands.Create("cheat_infinitearmor", [](CBasePlayer* player, const CommandArgs& args)
+		{ player->ToggleCheat(Cheat::InfiniteArmor); },
+		{.Flags = ClientCommandFlag::Cheat});
+
+	g_ClientCommands.Create("ent_find_by_classname", [](CBasePlayer* player, const CommandArgs& args)
+		{
+			if (args.Count() > 1)
+			{
+				int count = 0;
+				ALERT(at_console, "entindex - targetname - origin");
+
+				for (auto entity : UTIL_FindEntities()) {
+					if (!entity || !FStrEq(args.Argument(1), entity->GetClassname()))
+						continue;
+
+					ALERT(at_console, "%d - %s - {%f, %f, %f}", entity->entindex(), entity->GetTargetname(), entity->pev->origin.x, entity->pev->origin.y, entity->pev->origin.z);
+					count++;
+				}
+
+				ALERT(at_console, "%d entities having the classname: \"%s\"", count, args.Argument(1));
+			}
+			else
+			{
+				CLIENT_PRINTF(player->edict(), print_console, "usage: ent_find_by_classname <classname>\n");
+			} },
+		{.Flags = ClientCommandFlag::Cheat});
+
+	g_ClientCommands.Create("ent_find_by_targetname", [](CBasePlayer* player, const CommandArgs& args)
+		{
+			if (args.Count() > 1)
+			{
+				int count = 0;
+				ALERT(at_console, "entindex - classname - origin");
+
+				for (auto entity : UTIL_FindEntities()) {
+					if (!entity || !FStrEq(args.Argument(1), entity->GetClassname()))
+						continue;
+
+					ALERT(at_console, "%d - %s - {%f, %f, %f}", entity->entindex(), entity->GetTargetname(), entity->pev->origin.x, entity->pev->origin.y, entity->pev->origin.z);
+				}
+
+				ALERT(at_console, "%d entities having the targetname: \"%s\"", count, args.Argument(1));
+			}
+			else
+			{
+				CLIENT_PRINTF(player->edict(), print_console, "usage: ent_find_by_targetname <targetname>\n");
+			} },
+		{.Flags = ClientCommandFlag::Cheat});
+
+	g_ClientCommands.Create("ent_fire", [](CBasePlayer* player, const CommandArgs& args)
+		{
+			// TODO - Add "delay" like Source
+			if (args.Count() > 1)
+			{
+				USE_TYPE useType = USE_TOGGLE;
+				if (args.Count() > 2)
+				{
+					useType = std::clamp(static_cast<USE_TYPE>(atoi(args.Argument(2))), USE_OFF, USE_TOGGLE);
+				}
+
+				float value = 0.f;
+				if (args.Count() > 3)
+				{
+					value = atof(args.Argument(3));
+				}
+
+				FireTargets(args.Argument(1), player, player, useType, value);
+			}
+			else
+			{
+				CLIENT_PRINTF(player->edict(), print_console, "usage: ent_fire <targetname> [usetype] [value]\n");
+			} },
+		{.Flags = ClientCommandFlag::Cheat});
+
+	g_ClientCommands.Create("ent_list", [](CBasePlayer* player, const CommandArgs& args)
+		{
+			int edicts = 0;
+			int count = 0;
+			std::vector<const char*> entities;
+
+			for (auto entity : UTIL_FindEntities()) {
+				if (!entity)
+					continue;
+
+				if (entity->edict())
```

Comment:
Any entity that doesn't have an edict is bugged so this isn't really useful. I'd expect the game to crash before this condition becomes false.

Reply:
Removed everything "edicts count" related.